src/miner.cpp

```diff
@@ -251,7 +252,7 @@ CBlockTemplate* Mining::CreateNewBlock(const CChainParams& chainparams) const
             if (!IsFinalTx(tx, nHeight, nLockTimeCutoff))
                 continue;
 
-            const uint64_t maxSigOps = Policy::blockSigOpAcceptLimit(nBlockSize + nTxSize);
+            const uint64_t maxSigOps = Policy::blockSigOpAcceptLimit(nBlockSize + nTxSize - nCoinbaseReserveSize);
```

Comment:
nBlockSize is incremented later on in the loop. Line 269 is: `nBlockSize += nTxSize`.

This function really needs to be refactored away from being one giant loop :)